shm/Cargo.cpp

```diff
@@ -0,0 +1,31 @@
+#include "Cargo.hpp"
+
+Cargo::Cargo(std::string name, size_t amount, size_t basePrice)
+    : name_{name}, amount_{amount}, basePrice_{basePrice} {}
+
+
+Cargo& Cargo::operator+=(size_t amount) {
+    amount_ += amount;
+    return *this;
+}
+
+Cargo& Cargo::operator-=(size_t amount) {
+    amount_ -= amount;
+    return *this;
+}
```

Comment:
What happened when we reach the limit of capacity or drop below 0? You need to handle error cases. Maybe throw exception, maybe return `std::nullopt` or `nullptr` maybe set some specific value like `std::numeric_limits::min/max`.